flume-ng-node/src/test/java/org/apache/flume/node/TestAbstractConfigurationProvider.java

```diff
@@ -185,6 +185,36 @@ public void testSourceAndSinkThrowExceptionDuringConfiguration()
     Assert.assertTrue(config.getSinkRunners().size() == 0);
   }
 
+  @Test
+  public void testSinkSourceMismatchDuringConfiguration() throws Exception {
+    String agentName = "agent1";
+    String sourceType = "seq";
+    String channelType = "memory";
+    String sinkType = "avro";
+    Map<String, String> properties = getProperties(agentName, sourceType,
+        channelType, sinkType);
+    properties.put(agentName + ".sources.source1.batchSize", "100");
+    properties.put(agentName + ".sinks.sink1.batch-size", "100");
+    properties.put(agentName + ".sinks.sink1.hostname", "10.10.10.10");
+    properties.put(agentName + ".sinks.sink1.port", "1010");
+
+    MemoryConfigurationProvider provider =
+        new MemoryConfigurationProvider(agentName, properties);
+    MaterializedConfiguration config = provider.getConfiguration();
+    Assert.assertTrue(config.getSourceRunners().size() == 1);
+    Assert.assertTrue(config.getChannels().size() == 1);
+    Assert.assertTrue(config.getSinkRunners().size() == 1);
+
+    properties.put(agentName + ".sources.source1.batchSize", "1001");
+    properties.put(agentName + ".sinks.sink1.batch-size", "1001");
```

Comment:
Why is it set to 1001?
As far as I understand the point is to exceed the channel capacity, but it is 100 by default, so 101 would also work.
It would be more straightforward to set the channel capacity explicitly in the test instead of using the default one.

Reply:
agree